r/R/install-arrow.R

```diff
@@ -137,3 +136,93 @@ reload_arrow <- function() {
     message("Please restart R to use the 'arrow' package.")
   }
 }
+
+
+#' Create a source bundle that includes all thirdparty dependencies
+#'
+#' @param dest_file File path for the new tar.gz package. Defaults to
+#' `arrow_V.V.V_with_deps.tar.gz` in the current directory (`V.V.V` is the version)
+#' @param source_file File path for the input tar.gz package. Defaults to
+#' downloading the package from CRAN (or whatever you have set as the first in
+#' `getOption("repos")`)
+#' @return The full path to `dest_file`, invisibly
+#'
+#' This function is used for setting up an offline build. If it's possible to
+#' download at build time, don't use this function. Instead, let `cmake`
+#' download the required dependencies for you.
+#' These downloaded dependencies are only used in the build if
+#' `ARROW_DEPENDENCY_SOURCE` is unset, `BUNDLED`, or `AUTO`.
+#' https://arrow.apache.org/docs/developers/cpp/building.html#offline-builds
+#'
```

Comment:
```suggestion
#'
#' Note: If you're using RStudio Package Manager to download binary packages on
#' linux you shouldn't need to use this function. You can download the appropriate
#' binary from RStudio Package Manager, and transfer that to the offline computer.
#' If you still do want to make a source bundle with this function, make sure to
#' set the first repo in `options("repos")` to be a mirror that contains source
#' packages (that is: something other than the RStudio Package Manager binary 
#' mirror URLs). 
#'
```

How about adding this to the docs. We could do something like sniff to see if `libarrow.so` is in the package downloaded, but that is fragile + this is a pretty unique workflow.

Reply:
While we're at it, should we mention other binary platforms?

```
#' Note: If you're using binary packages, e.g. from RStudio Package Manager on
#' Linux or the standard CRAN binaries on Windows or Mac, you shouldn't need to
#' use this function. You can download the appropriate binary from your package
#' repository, and transfer that to the offline computer.
#' If you still want to make a source bundle with this function, make sure to
#' set the first repo in `options("repos")` to be a mirror that contains source
#' packages (that is: something other than the RSPM binary mirror URLs).
#' Any OS can create the source bundle, but it cannot be installed on Windows.
#' (Instead, use a standard Windows binary package.)
```

Reply:
Sure, I'm a little bit less worried about those since they will either just work (on MacOS) or have other issues (windows). But adding them for completeness is good. I've re-arranged the wording a little bit to be more general part first ("if you've got access to full-featured binaries, use those!") and then the special note about RSPM/linux second (since that's the only platform where `type="source"` isn't respected)

```
#' If you're using binary packages you shouldn't need to use this function. You 
#' should download the appropriate binary from your package repository, transfer 
#' that to the offline computer, and install that. Any OS can create the source 
#' bundle, but it cannot be installed on Windows. (Instead, use a standard 
#' Windows binary package.)
#'
#' Note if you're using RStudio Package Manager on Linux: If you still want to 
#' make a source bundle with this function, make sure to set the first repo in 
#' `options("repos")` to be a mirror that contains source packages (that is: 
#' something other than the RSPM binary mirror URLs). 
```

Reply:
Thanks!